custom_components/roborock/config_flow.py

```diff
@@ -149,21 +157,26 @@ async def _login(self, code) -> UserData:
 
 
 def set_nested_dict(data: dict, key_string: str, value):
+    """Set nested dict."""
     here = data
     keys = key_string.split(".")
     for key in keys[:-1]:
         here = here.setdefault(key, {})
     here[keys[-1]] = value
 
+
 def get_nested_dict(data: dict, key_string: str, default=None):
+    """Get nested dict."""
     here = data
     keys = key_string.split(".")
     for key in keys:
         here = here.get(key)
-        if not here:
+        # Check for None type since CONF_INCLUDE_SHARED can return False
+        if here is None:
```

Comment:
Holly s***. All that store_type and show_type stupid struct had to be done probably because of this bug 🤦🏻‍♂️